#panda:                    #useful if you use a namespace for the robot
    # Publish joint states
    joint_state_controller:
      type: joint_state_controller/JointStateController
      publish_rate: 50

    cartesian_impedance_controller:
      type: panda_3dbioprint_real/CartesianImpedanceController
      arm_id: panda
      joints:
        - panda_joint1
        - panda_joint2
        - panda_joint3
        - panda_joint4
        - panda_joint5
        - panda_joint6
        - panda_joint7

      end_effector_params:
        - panda_pos_x
        - panda_pos_y
        - panda_pos_z
        - panda_orient_x
        - panda_orient_y
        - panda_orient_z

      constraints:
        goal_time: 2.0

      state_publish_rate: 25

    gravity_compensation_controller:
      type: panda_3dbioprint_real/GravityCompensationController
      arm_id: panda
      joints:
        - panda_joint1
        - panda_joint2
        - panda_joint3
        - panda_joint4
        - panda_joint5
        - panda_joint6
        - panda_joint7

      constraints:
        goal_time: 2.0

      state_publish_rate: 25